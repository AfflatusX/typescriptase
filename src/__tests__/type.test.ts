import { type1, type2, type3, type4, type5 } from "./examples";

test(
  "type1",
  async (): Promise<void> => {
    expect(type1.print({
      name: "type1",
      path: "src/__tests__/type.test.ts",
    }))
      .toMatchSnapshot();
  },
);

test(
  "type2",
  async (): Promise<void> => {
    expect(type2.print({
      name: "type2",
      path: "src/__tests__/type.test.ts",
    }))
      .toMatchSnapshot();
  },
);

test(
  "type3",
  async (): Promise<void> => {
    expect(type3.print({
      name: "type3",
      path: "src/__tests__/type.test.ts",
    }))
      .toMatchSnapshot();
  },
);

test(
  "type4",
  async (): Promise<void> => {
    expect(type4.print({
      name: "type4",
      path: "src/__tests__/type.test.ts",
    }))
      .toMatchSnapshot();
  },
);

test(
  "type5",
  async (): Promise<void> => {
    expect(type5.print({
      name: "type5",
      path: "src/__tests__/type.test.ts",
    }))
      .toMatchSnapshot();
  },
);
