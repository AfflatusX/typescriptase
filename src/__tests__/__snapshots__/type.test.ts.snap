// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`type1 1`] = `
"type Ty1 = number;
"
`;

exports[`type2 1`] = `
"export type Ty2 = string | null;
"
`;

exports[`type3 1`] = `
"export type Ty3 = Ty2 | Ty1;
"
`;
